### PR TITLE
PP-477: Server crash due to invalid memory access

### PR DESCRIPTION
**Pull request by vinodchitrali**
#### Issue-ID

* **[PP-477](https://pbspro.atlassian.net/browse/PP-477)**

#### Problem description
* Server crash due to invalid memory access 

#### Cause / Analysis
* Pointer to the batch_request is stored in the job, and is used by time out function. However, when the connection is closed, the batch request is found and the connection is set to -1. timeout_rerun_request uses connection which was set to -1 and try to actual connection and hence invalid memory access

#### Solution description
* return from connection_find_actual_index function if the sock is -1 and in timeout_rerun_request the conn_idx is retrieved prior to reply_send 

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [x] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by rampranesh** [comment]
Hi Vinod,
      Changes looks good, I sign off.

Thanks,
Rampranesh M

File: src/server/req_rerun.c [hunk 1]
```diff
@@ -358,16 +358,15 @@ timeout_rerun_request(struct work_task *pwt)
...
 	reply_text(pjob->ji_rerun_preq, PBSE_INTERNAL,
 		"rerunning job has timed out");
 
 	/* clear no-timeout flag on connection */
-	if (pjob->ji_rerun_preq->rq_conn != PBS_LOCAL_CONNECTION) {
-		conn_idx = connection_find_actual_index(pjob->ji_rerun_preq->rq_conn);
-		if (conn_idx != -1)
-			svr_conn[conn_idx].cn_authen &= ~PBS_NET_CONN_NOTIMEOUT;
-	}
+	if (conn_idx != -1)
```
Comment (arungrover) [inline src/server/req_rerun.c]: conn_idx is not initialized... So if it turns out to be a local connection then also you may end up getting -1 in conn_idx variable.
Can you please explain a little why this change is needed?

Comment (subhasisb) [inline src/server/req_rerun.c]: Yes, conn_idx must be initialized for this to work properly. Good catch @arungrover

Comment (vinodchitrali) [inline src/server/req_rerun.c]: fixed

**Comment by arungrover** [comment]
Vinod, please rebase again and have you will need one more reviewer to review it again since the code has changed.

**Comment by vinodchitrali** [comment]
done

**Comment by suresh-thelkar** [comment]
Code changes look good to me.

**Comment by arungrover** [comment]
@subhasisb your approval is needed for the ticket to move forward

**Comment by subhasisb** [comment]
All looks good. Updating code and merging.
